.pre-commit-config.yaml

```diff
@@ -0,0 +1,50 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+default_language_version:
+    python: python3.7
+default_stages: [commit, push]
+fail_fast: false
+repos:
+-   repo: https://github.com/psf/black
+    rev: 21.12b0
+    hooks:
+    -   id: black
+        args: [--safe, --config=pyproject.toml]
+
+-   repo: https://github.com/pre-commit/pre-commit-hooks
```

Comment:
pre-commit allows a multitude of hooks to run when commiting code.
See https://pre-commit.com/hooks.html for all supported hooks.